dashboard/views.py

```diff
@@ -321,60 +326,69 @@ def resource_access_within_week(request, course_id=0):
     df.drop_duplicates(inplace=True)
 
-    resource_id_name=df["resource_id_name"].unique()
+
+    resource_id_type=df["resource_id_type"].unique()
 
     #df.reset_index(inplace=True)
 
     # zero filled dataframe with resource name as row name, and grade as column name
-    output_df=pd.DataFrame(0.0, index=resource_id_name, columns=[GRADE_A, GRADE_B, GRADE_C, GRADE_LOW, NO_GRADE_STRING, RESOURCE_TYPE_STRING])
-    output_df=output_df.rename_axis('resource_id_name')
+    output_df=pd.DataFrame(0.0, index=resource_id_type, columns=['r_id', 'r_name', GRADE_A, GRADE_B, GRADE_C, GRADE_LOW, NO_GRADE_STRING, RESOURCE_TYPE_STRING])
+    output_df=output_df.rename_axis('resource_id_type')
     output_df=output_df.astype({RESOURCE_TYPE_STRING: str})
+    output_df=output_df.astype({'r_name': str})
+    output_df=output_df.astype({'r_id': str})
+
 
     for index, row in df.iterrows():
         # set value
-        output_df.at[row['resource_id_name'], row['grade']] = row['percent']
-        output_df.at[row['resource_id_name'], RESOURCE_TYPE_STRING] = row[RESOURCE_TYPE_STRING]
+        output_df.at[row['resource_id_type'], row['grade']] = row['percent']
+        output_df.at[row['resource_id_type'], RESOURCE_TYPE_STRING] = row[RESOURCE_TYPE_STRING]
+        output_df.at[row['resource_id_type'], 'r_name'] = row['name']
+        output_df.at[row['resource_id_type'], 'r_id'] = row['resource_id']
     output_df.reset_index(inplace=True)
 
     # now insert person's own viewing records: what resources the user has viewed, and the last access timestamp
-    selfSqlString = f"""select CONCAT(r.resource_id, ';', r.name) as resource_id_name, count(*) as self_access_count, max(a.access_time) as self_access_last_time 
+    selfSqlString = f"""
+                    select
+                    r.resource_id as resource_id,
+                    r.resource_type as resource_type,
+                    CONCAT(r.resource_id, r.resource_type) as resource_id_type,
+                    r.name as name,
+                    count(*) as self_access_count,
+                    max(a.access_time) as self_access_last_time 
                     from resource_access a, user u, resource r 
                     where a.user_id = u.user_id 
                     and a.resource_id = r.resource_id 
                     and u.sis_name=%(current_user)s 
                     and a.course_id = %(course_id)s
                     and a.course_id = u.course_id
-                    group by CONCAT(r.resource_id, ';', r.name)"""
+                    group by r.resource_id, r.resource_type, r.name"""
     logger.debug(selfSqlString)
     logger.debug("current_user=" + current_user)
 
     selfDf= pd.read_sql(selfSqlString, conn, params={"current_user":current_user, "course_id": course_id})
 
-    output_df = output_df.join(selfDf.set_index('resource_id_name'), on='resource_id_name', how='left')
+    selfDf.drop(columns=['resource_type'], inplace=True)
```

Comment:
You could just exclude this from the query (On line 372) rather than dropping it in the dataframe. You aren't using resource_type anywhere else, only the reference to r.resource_type on 383.